packages/wallet/src/redux/protocols/funding/player-b/reducer.ts

```diff
@@ -115,19 +179,58 @@ function strategyApproved(
   const message = sendStrategyApproved(opponentAddress, processId);
 
   const channelState = selectors.getChannelState(sharedData, state.targetChannelId);
-  const { protocolState: fundingState, sharedData: newSharedData } = initializeIndirectFunding(
-    processId,
-    channelState,
-    sharedData,
-  );
-  if (indirectFundingStates.isTerminal(fundingState)) {
-    console.error('Indirect funding strate initialized to terminal state.');
-    return handleFundingComplete(state, fundingState, newSharedData);
+
+  if (state.strategy === 'ExistingChannelStrategy') {
+    const existingLedgerChannel = selectors.getExistingLedgerChannelForParticipants(
+      sharedData,
+      state.ourAddress,
+      state.opponentAddress,
+    );
+    if (
+      !existingLedgerChannel ||
+      getLastCommitment(existingLedgerChannel).commitmentType !== CommitmentType.App
+    ) {
+      throw new Error(
```

Comment:
Maybe in the future we want to handle this state more gracefully, right?

Reply:
Yeah, I think there's more work to be done around this. There should be a negotiation between two players on whether they have the existing ledger channel and if they want to use it.